lib/qmp.ml

```diff
@@ -215,10 +228,18 @@ let message_of_string str =
       let fdset_id = json |> U.member "fdset-id" |> U.to_int in
       {fd; fdset_id}
     in
+    let qom json =
+      let mem_of k x = x |> U.member k |> U.to_string in
+      json |> U.convert_each (fun x -> {name = (mem_of "name" x);ty = (mem_of "type" x)})
+    in
     let result =
       let return = json |> U.member "return" in
       return |> function
-      | `List  _ -> return |> name_list |> fun x -> Name_list x
+      | `List  _ -> (return |> U.convert_each (fun x -> x |> U.keys) |> List.flatten |> function
+        | x when x |> subset_of ["name"; "type"] -> return |> qom |> fun x -> Qom x
```

Comment:
```ocaml
| x when subset_of ["name"; "type"] x -> ...
| x when subset_of ["name"] x -> ...
| _ -> ...
```

Reply:
this will catch error when build it. 
File "lib/qmp.ml", line 242, characters 17-43:
Error: This expression has type string list -> bool
       but an expression was expected of type bool

Reply:
Ad parentheses then
```ocaml
| x when (subset_of ["name"; "type"] x) -> ...
| x when (subset_of ["name"] x) -> ...
| _ -> ...
```
I think it's more readable then interspersing pipes and arrows

Reply:
OK , got that